dew/git.py

```diff
@@ -38,14 +38,13 @@ def checkout(repo: git.Repo, origin: git.Remote, head_name: str, ref: str) -> No
     for submodule in repo.submodules:
         print(f'Processing submodule {submodule.name}', flush=True)
 
-        # Calling git directly for own submodules since using relative path is not working in gitpython
-        # see https://github.com/gitpython-developers/GitPython/issues/730
-        if submodule.url[0:3] == '../':
-            repo.git.submodule('init', submodule.name)
-            repo.git.submodule('update', submodule.name)
-        # For external submodules we can use the update function of gitpython
-        else:
-            submodule.update(init=True)
+        # Calling git directly for own submodules since using relative path is not working in gitpython
+        # see https://github.com/gitpython-developers/GitPython/issues/730
+        if submodule.url[0:3] == '../':
+            repo.git.submodule('update', '--init', '--recursive', submodule.name)

```

Comment:
Sometimes I wonder why I even bother with using gitpython when a lot of the time it is implemented through git invocations. Seems like using subprocess.call with git commands would be more useful and stable. As long as we version check git it would work nicely I think.

Reply:
GitPython makes introspection easier, but I'm not sure whether or not that is something dew needs. It seems like it could go without. In any case, it would be easy to replace it with your own subprocess-based driver module.